streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -93,7 +93,9 @@ static void registerStateStores(final Logger log,
         }
 
         final TaskId id = stateMgr.taskId();
-        if (!stateDirectory.lock(id)) {
+        if (!stateDirectory.canTryLock(id, System.currentTimeMillis())) {
```

Comment:
If we enter this branch, the next line that will be printed is "acquired state directory lock". But this is not what we want right? Don't we have to leave this method somehow?

We should also test this behavior in `StateManagerUtilTest`.